### DISCUSSION TITLE
Connect PowerBI to Supabase and allow refresh in PowerBI Web Services

### DISCUSSION THREAD

**Discussion by jlmingo** [supabase/supabase#29020] [Questions]
Hi,
I know there is a thread explaining how to connect powerbi to Supabase using the ODBC driver and a gateway. It works just fine. I can also refresh the dataset as long as my computer (where the gateway is) is turned on.

However, I would like to leave the dataset in the web and have it refreshed automatically even if my computer is turned off. I tried to make a connection using the web API, which works in PowerBI Desktop but won't refresh in the web.

Once I publish the dataset to PowerBI, it asks for credentials and won't accept my Supabase user and password. I am missing something when making the request in powerbi (maybe the headers)? Any other materials or guidance to check this use case? Below is the code:

let

// Define the API URL

apiUrl = "https://xxx.supabase.co/rest/v1/dim_project",

// Set up the HTTP headers with both the API key and Authorization Bearer token

headers = [

#"apikey" = "xxx",

#"Authorization" = "Bearer xxx"

],

// Send the GET request to the API with the specified headers

response = Web.Contents(apiUrl, [

Headers = headers

]),

// Parse the JSON response

json = Json.Document(response),

// Convert JSON list to table

jsonTable = Table.FromList(json, Splitter.SplitByNothing(), null, null, ExtraValues.Error),

#"Expanded Column1" = Table.ExpandRecordColumn(jsonTable, <some columns>],

#"Removed Other Columns" = Table.SelectColumns(#"Expanded Column1",{some columns})

in

#"Removed Other Columns"

**Comment by salomaoparkour** [comment]
Any luck with this, after 2 years? @jlmingo
